README.md

```diff
@@ -32,27 +32,179 @@ npm install
+
+example: 
 ```js
-const {createAuthzHeader, createSignatureString} = require('http-signature-header');
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 1 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: ['one']}],
+  ]),
+  // this will add the params to all the signatures
+  params: {alg: 'foo'}
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo"
+```
 
-const requestOptions = {
-  url,
-  method: 'POST',
-  headers
-}
-const includeHeaders = ['expires', 'host', '(request-target)'];
-const plaintext = createSignatureString({includeHeaders, requestOptions});
+example: 
+```js
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 2 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: [{value: 'one'}]}],
+    ['sig2', {coveredContent: [{value: 'two'}]}]
+  ]),
+  // this will add the params to all the signatures
+  params: {alg: 'foo'}
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo", sig2=("two");alg="foo"
+```
+example: 
+```js
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 2 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: [{value: 'one'}], params: {alg: 'foo'}}],
+  // sig2 uses a different signing algorithm `bar`
+    ['sig2', {coveredContent: [{value: 'two'}], params: {alg: 'bar'}}]
+  ]),
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo", sig2=("two");alg="bar"
+```
 
-const data = new TextEncoder().encode(plaintext);
-const signature = base64url.encode(await signer.sign({data}));
+`createSignatureInputString`
 
-const Authorization = createAuthzHeader({
-  includeHeaders,
-  keyId: signer.id,
-  signature
+```js
+const date = new Date().toUTCString();
+const httpMessage = {
+  headers: {date},
+  method: 'GET',
+  url: 'https://example.com',
+};
+const coveredContent = [{value: 'one'}];
+const params = {alg: 'foo'};
+// this turns your coveredContent etc into SF Items
+const stringToSign = httpSignatureHeader.createSignatureInputString({
+  coveredContent,
+  params,
+  httpMessage
 });
+const data = new TextEncoder().encode(stringToSign);
+// invocationSigner being a key
+const signature = await invocationSigner.sign({data});
+```
+
+`createSignatureHeader` is much easier. Sign the signing string for your signature
+and pass it as a `Uint8Array`.
+
+example: 
+```js
+const signature = new Uint8Array([1, 2, 3])
+const sigs = {sig1: signature};
+const dict = httpSignatureHeader.createSignatureHeader(sigs);
+console.log(dict);
+//'sig1=:AQID:'
 ```
 
+
+`parseRequest` takes in a request that is expected to be similar to
+a request object in `express.js`. It then parses the `Signature-Input` header and returns
+a Map with the signature id and then an object with everything you need to sign:
```

Comment:
```suggestion
The `parseRequest` API takes in a request object similar to a request object in `express.js`. 
It then parses the `Signature-Input` header and returns
a Map with the signature id and then an object with everything needed to create a signed header:
```
I was confused by the meaning of "to sign", I changed it above to what I think is meant by those words.